src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -50,15 +55,35 @@ public ModelManager(ReadOnlyAddressBook addressBook, UserPrefs userPrefs) {
 
         this.addressBook = new AddressBook(addressBook);
         filteredParcels = new FilteredList<>(this.addressBook.getParcelList());
+        updatedDeliveredAndUndeliveredList();
+        activeFilteredList = filteredUndeliveredParcels;
     }
 
     public ModelManager() {
         this(new AddressBook(), new UserPrefs());
     }
 
+    /**
+     * Updates Delivered and UndeliveredParcelList and resets the Active List to the correct reference
+     */
+    private void updatedDeliveredAndUndeliveredList() {
+        boolean isActiveDelivered = activeFilteredList == filteredDeliveredParcels;
```

Comment:
Add a comment to to say that you are checking for reference equality and so that people will know that what you did was intended